breathecode/payments/tests/urls/tests_pay.py

```diff
@@ -848,6 +848,128 @@ def test_with_chosen_period__amount_set(bc: Breathecode, client: APIClient):
     )
 
 
+def test_with_chosen_period__amount_set_with_conversion_info(bc: Breathecode, client: APIClient):
+    bag = {
+        'token': 'xdxdxdxdxdxdxdxdxdxd',
+        'expires_at': UTC_NOW,
+        'status': 'CHECKING',
+        'type': 'BAG',
+        **generate_amounts_by_time(),
+    }
+    chosen_period = random.choice(['MONTH', 'QUARTER', 'HALF', 'YEAR'])
+    amount = get_amount_per_period(chosen_period, bag)
+
+    plan = {'is_renewable': False}
+
+    model = bc.database.create(user=1, bag=bag, academy=1, currency=1, plan=plan, service_item=1)
+    client.force_authenticate(user=model.user)
+
+    url = reverse_lazy('payments:pay')
+    data = {
+        'token': 'xdxdxdxdxdxdxdxdxdxd',
+        'chosen_period': chosen_period,
+        'conversion_info': {
+            'landing_url': '/home'
+        },
+    }
+    response = client.post(url, data, format='json')
+
+    json = response.json()
+    expected = get_serializer(bc, model.currency, model.user, data={'amount': math.ceil(amount)})
+
+    assert json == expected
+    assert response.status_code == status.HTTP_201_CREATED
+
+    assert bc.database.list_of('payments.Bag') == [{
+        **bc.format.to_dict(model.bag),
+        'token': None,
+        'status': 'PAID',
+        'expires_at': None,
+        'chosen_period': chosen_period,
+    }]
+    assert bc.database.list_of('payments.Invoice') == [
+        format_invoice_item({
+            'amount': math.ceil(amount),
+            'stripe_id': '1',
+        }),
+    ]
+    assert bc.database.list_of('authenticate.UserSetting') == [
+        format_user_setting({'lang': 'en'}),
+    ]
+
+    bc.check.queryset_with_pks(model.bag.plans.all(), [1])
+    bc.check.queryset_with_pks(model.bag.service_items.all(), [1])
+    assert tasks.build_subscription.delay.call_args_list == [call(1, 1, conversion_info="{'landing_url': '/home'}")]
```

Comment:
It should be fine to have tests with `build_plan_financing` and `build_free_subscription` too